src/UnwindCursor.hpp

```diff
@@ -2760,7 +2760,15 @@ int UnwindCursor<A, R>::stepThroughSigReturn(Registers_arm64 &) {
     _registers.setRegister(UNW_AARCH64_X0 + i, value);
   }
   _registers.setSP(_addressSpace.get64(sigctx + kOffsetSp));
-  _registers.setIP(_addressSpace.get64(sigctx + kOffsetPc));
+
+  // The +1 story is the same as in DwarfInstructions::stepWithDwarf()
+  // (search for "returnAddress + cieInfo.isSignalFrame" or "Return address points to the next instruction").
+  // This is probably not the right place for this because this function is not necessarily used
+  // with DWARF. Need to research whether the other unwind methods have the same +-1 situation or
+  // are off by one.
+  pint_t returnAddress = _addressSpace.get64(sigctx + kOffsetPc);
+  _registers.setIP(returnAddress + 1);
```

Comment:
AFAIU this patch it is not required with the last version of mine initial patch - https://github.com/llvm/llvm-project/pull/92291/files

Since now this `+1` is in `setInfoBasedOnIPRegister` which is called for arm64 as well, so let's try to reapply proper version of the patch - https://github.com/ClickHouse/libunwind/pull/31